Sources/StreamChat/Models/Channel+Sorting.swift

```diff
@@ -0,0 +1,166 @@
+            } else if let lDouble = lhs as? Double, let rDouble = rhs as? Double {
+                return isAscending ? lDouble < rDouble : lDouble > rDouble
+            } else if let lDate = lhs as? Date, let rDate = rhs as? Date {
+                return isAscending ? lDate < rDate : lDate > rDate
+            } else if let lBool = lhs as? Bool, let rBool = rhs as? Bool {
+                return isAscending ? !lBool && rBool : lBool && !rBool
+            }
+
+            return false
+        }
+
+        return channels.sorted { lhs, rhs in
+            for sort in sorting {
+                let lhsValue = lhs[keyPath: sort.keyPath]
+                let rhsValue = rhs[keyPath: sort.keyPath]
+
+                if sort.isAscending, evaluate(lhs: lhs, rhs: rhs, isAscending: sort.isAscending) {
+                    return true
+                } else if !sort.isAscending, evaluate(lhs: lhs, rhs: rhs, isAscending: sort.isAscending) {
+                    return true
+                } else {
+                    continue
+                }
+            }
+
+            return false
+        }
+    }
+}
+
+extension ChatChannel {
+    // This method is meant to only be used for sorting functionality.
+    subscript(dynamicMember member: String) -> String? {
+        switch member {
+        case "cid":
+            return cid.rawValue
+        case "name":
+            return name
+        case "type":
+            return type.rawValue
+        case "lastMessageAt":
+            return lastMessageAt?.sortValue
+        case "createdBy":
+            return createdBy?.id
+        case "createdAt":
+            return createdAt.sortValue
+        case "updatedAt":
+            return updatedAt.sortValue
+        case "deletedAt":
+            return deletedAt?.sortValue
+        case "truncatedAt":
+            return truncatedAt?.sortValue
+        case "isHidden":
+            return isHidden.sortValue
+        case "isFrozen":
+            return isFrozen.sortValue
+        case "isMuted":
+            return isMuted.sortValue
+        case "isDeleted":
+            return isDeleted.sortValue
+        case "isDirectMessageChannel":
+            return isDirectMessageChannel.sortValue
+        case "isUnread":
+            return isUnread.sortValue
+        case "memberCount":
+            return memberCount.sortValue
+        case "watcherCount":
+            return watcherCount.sortValue
+        case "membership":
+            return membership?.memberRole.rawValue
+        case "unreadCount":
+            return unreadCount.messages.sortValue
+        case "lastMessageFromCurrentUser":
+            return lastMessageFromCurrentUser?.createdAt.sortValue
+        case "team":
+            return team
+        default:
+            break
+        }
+
+        // Does not work with nested values (eg. "scope.score" -> `extraData: { scope : { score : 1 }}`)
+        if let value = extraData[member] {
+            return value.sortValue
+        }
```

Comment:
We could support nested values actually 🤔 On the extraData, we can check if the key contains any ".", if yes we can nest it. Example: `user.name` we would split by `.` and convert it to `extraData["user"]?extraData["name"]`